### DISCUSSION TITLE
Ideas for lua scripting

### DISCUSSION THREAD

**Discussion by LexxxyT** [Mogli12/AIVehicleExtension#298] [Ideas]
Hi all, Ok I'm no expert with lua scripting but you are amazing so I was wondering if you would be able to implement some scripts. You can take credit for them, I don't mind. I was wondering if you could create a script to make it that when you park a vehicle in a "repair" spot, it takes a few hours/days to repair depending on the damage. Also, I tried to make a script to change the color and amount of the exhaust smoke depending on the age and damage of the vehicle. I was trying to make the exhaust turn from almost clear to a distinct bluish tint to simulate oil burning... I couldn't do it but if you are able to take a look that would be great! Cheers! Lexxy.

**Comment by Dzi4d3k** [comment]
@LexxxyT 
Did you checked AutoDrive for repair functionality:
https://github.com/Stephan-S/FS22_AutoDrive

> https://github.com/Stephan-S/FS22_AutoDrive/wiki/Release-Notes-for-AutoDrive-english

**Reply by LexxxyT** [reply]
Yes I run the autodrive mod already. Repair functionality does work but as I said I'm looking for something to simulate the repair actually taking time to complete... As in real life, repairs don't happen instantly.

**Comment by Dzi4d3k** [comment]
Oh I see what you mean, but I think is a good idea for standalone mod like Increased Unloading Time mod by coolfarmer on Giants' ModHub.

**Reply by LexxxyT** [reply]
Yes that's what I was thinking. Thanks for pointing out the increased unloading time mod too. I've just installed it and it's much better now.

**Reply by jason0611** [reply]
There was a mod for FS19, "Extended Vehicle Maintanence", that offered exactly this. Maybe it will make it into FS22 sometimes?

**Reply by LexxxyT** [reply]
Yes this is the sort of thing I was looking for. I hope someone does make a conversion of this mod.

**Reply by Dzi4d3k** [reply]
Another idea would be to ask author of recently added mod:
https://farming-simulator.com/mod.php?lang=en&country=gb&mod_id=230629&title=fs2022
to add additional settings for configuring repair duration.
